metacatalog/ext/standard_export/extension.py

```diff
@@ -0,0 +1,109 @@
+    to the folder location specified in ``path``.
+
+    """
+    @classmethod
+    def init_extension(cls):
+        # wrapper which calls StandardsExportExtension.iso19115_export
+        def wrapper_entry(self: Entry, config_dict: dict):  
+            return StandardsExportExtension.iso19115_export(entry_or_resultset=self, config_dict=config_dict)
+        
+        # iso19115_export docstring and name for wrapper function
+        wrapper_entry.__doc__ = StandardsExportExtension.iso19115_export.__doc__
+        wrapper_entry.__name__ = StandardsExportExtension.iso19115_export.__name__
+        
+        # add wrapper to Entry model
+        Entry.export_iso19115 = wrapper_entry
+
+
+    @classmethod
+    def iso19115_export(cls, entry_or_resultset: Union[Entry, ImmutableResultSet], config_dict: dict):
+        """
+        Export a :class:`Entry <metacatalog.models.Entry>` or 
+        :class:`ImmutableResultSet <metacatalog.util.results.ImmutableResultSet>` to XML in 
+        ISO 19115 standard.
+        Repeatable information input is always a list, as we can loop over the lists in the
+        jinja ISO 19115 template.
+        Always returns an :class:`ElementTree <xml.etree.ElementTree.ElementTree>` object.
+
+        This function is added as a method to :class:`Entry <metacatalog.models.Entry>`
+        when the extension is activated. 
+
+        .. versionadded:: 0.7.7
+
+        Parameters
+        ----------
+        entry_or_resultset : Union[Entry, ImmutableResultSet]
+            The entry instance to be exported
+        config_dict : dict
+            Configuration dictionary, containing information about the data provider.
+            Mandatory (nested) keys and type of values:
+            * contact
+                * organisationName: str
+                * deliveryPoint: str
+                * city: str
+                * administrativeArea: str
+                * postalCode: str
+                * country: str
+                * electronicMailAddress: list(str)
+                * linkage: str
+                * linkage_name: str
+                * linkage_description: str
+            * publisher
+                * organisation_name: str
+        
+        Returns
+        ----------
+        xml : xml.etree.ElementTree.ElementTree
+            The :class:`ElementTree <xml.etree.ElementTree.ElementTree>` object
+            representing the XML ElementTree in Python.
+
+        Notes
+        -----
+        The content of the file will be created using a 
+        :class:`ImmutableResultSet <metacatalog.utils.results.ImmutableResultSet>`.
+        This will lazy-load sibling Entries and parent groups as needed for
+        a useful Metadata export.
+
+        """
+        # get necessary input parameters from ImmutableResultSet for ISO export
+        iso_input = _parse_iso_information(entry_or_resultset)
+
+        # get initialized jinja template
+        template = _init_iso19115_jinja()
```

Comment:
The user should be able to pass in an argument to this function in order to set a custom template.